### PR TITLE
Remove unnecessary rescale boolean variable from dropout layer

### PR DESCRIPTION
**Pull request by sourabhvarshney111**
This is towards completion of PR #1282. I have done the style fixes.

### CODE REVIEW

**Comment by sourabhvarshney111** [comment]
@zoq I think I have fixed the comments in #1282. Review whenever you are free.

File: src/mlpack/methods/ann/layer/dropout.hpp [hunk 1]
```diff
@@ -47,21 +47,17 @@ namespace ann /** Artificial Neural Network. */ {
  * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
  *         arma::sp_mat or arma::cube).
  */
-template <
-    typename InputDataType = arma::mat,
-    typename OutputDataType = arma::mat
->
+template<typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat>
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/dropout.hpp]: I don't see any need to change from before, but if you do change the style here, can you line up the `typename OutputDataType` with the `typename InputDataType` please?

File: src/mlpack/methods/ann/layer/dropout_impl.hpp [hunk 2]
```diff
@@ -38,23 +37,14 @@ void Dropout<InputDataType, OutputDataType>::Forward(
...
-    else
-    {
-      output = input * scale;
-    }
-  }
-  else
-  {
+  if (deterministic) {
+    output = input;
+  } else {
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/dropout_impl.hpp]: Can you fix the style here please?

```
if (deterministic)
{
  output = input;
}
else
{
  ...
}
```

Thanks!

File: src/mlpack/methods/ann/layer/dropout.hpp (hunk 1 above)
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/dropout.hpp]: Can you line up the `typename OutputDataType` with the `typename InputDataType`?  I think it is a little more readable that way.  (I'm sorry, I know it is a pedantic comment.)

File: src/mlpack/methods/ann/layer/dropout_impl.hpp (hunk 2 above)
Comment (sourabhvarshney111) [reply src/mlpack/methods/ann/layer/dropout_impl.hpp]: Forgot that. Done :)

File: src/mlpack/methods/ann/layer/dropout.hpp (hunk 1 above)
Comment (sourabhvarshney111) [reply src/mlpack/methods/ann/layer/dropout.hpp]: Done :+1:

**Comment by dakshitagrawal97** [comment]
Thanks @sourabhvarshney111 for taking this PR towards completion!  Appreciate it. :)

File: src/mlpack/methods/ann/layer/dropout.hpp [hunk 3]
```diff
@@ -85,22 +81,26 @@ class Dropout
...
   //! Get the input parameter.
-  InputDataType const& InputParameter() const { return inputParameter; }
+  InputDataType const& InputParameter() const { return inputParameter;  }
+
   //! Modify the input parameter.
   InputDataType& InputParameter() { return inputParameter; }
 
   //! Get the output parameter.
   OutputDataType const& OutputParameter() const { return outputParameter; }
+
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/dropout.hpp]: usually we group those two functions together.

Comment (sourabhvarshney111) [reply src/mlpack/methods/ann/layer/dropout.hpp]: You mean the call by value and call by reference ones @zoq ?

Comment (sourabhvarshney111) [reply src/mlpack/methods/ann/layer/dropout.hpp]: You mean the call by value and call by reference ones @zoq ?

Comment (zoq) [reply src/mlpack/methods/ann/layer/dropout.hpp]: yes

**Comment by sourabhvarshney111** [comment]
@zoq Done :+1:

**Comment by zoq** [comment]
@dakshitagrawal97 @sourabhvarshney111 thanks for the great contribution.

**Pull request merged**